forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -1640,9 +1640,15 @@ public static int xCount(Card c, final String s, final CardTraitBase ctb) {
                 return doXMath(calculateAmount(c, sq[v ? 1 : 2], ctb), expr, c, ctb);
             }
 
+            SpellAbility sa = null;
             if (ctb instanceof SpellAbility) {
-                final SpellAbility sa = (SpellAbility) ctb;
+                sa = (SpellAbility) ctb;
+            } else if (sq[0].contains("xPaid") && ctb instanceof TriggerReplacementBase) {
```

Comment:
What exactly is the reason for this?

Reply:
I need the better logic for the "if X is 5 or more" ravenous check or it still draws

Reply:
i will need to think of a better way, but i think this is good enough for now